taipy/gui/gui.py

```diff
@@ -452,42 +451,58 @@ def __process_content_provider(self, state: State, path: str, query: t.Dict[str,
                 if find_spec("plotly") and find_spec("plotly.graph_objs"):
                     from plotly.graph_objs import Figure as PlotlyFigure  # type: ignore[reportMissingImports]
 
-                    if isinstance(content, PlotlyFigure):
-
-                        def get_plotly_content(figure: PlotlyFigure):
-                            return figure.to_html()
-
-                        Gui.register_content_provider(PlotlyFigure, get_plotly_content)
+                    if isinstance(content, (PlotlyFigure, dict)):
+                        def get_plotly_content(content):
+                            mapbox_token = os.environ.get("MAPBOX_TOKEN")
+                            if mapbox_token is None:
+                                raise ValueError("MAPBOX_TOKEN environment variable is not set")
+                            if isinstance(content, PlotlyFigure):
+                                content.update_layout(mapbox_style="mapbox://styles/mapbox/streets-v11",
+                                mapbox_center_lat=37.7749, mapbox_center_lon=-122.4194,
+                                mapbox_zoom=12, mapbox_access_token=mapbox_token)
+                                return content.to_html()
+                            elif isinstance(content, dict):
+                                import plotly.graph_objects as go
+                                import json
+
+                                fig = go.Figure(data=content['data'], layout=content['layout'])
+                                fig.update_layout(mapbox_style="mapbox://styles/mapbox/streets-v11",
+                                mapbox_center_lat=37.7749, mapbox_center_lon=-122.4194,
+                                mapbox_zoom=12, mapbox_access_token=mapbox_token)
+                                html = fig.to_html(include_plotlyjs='cdn')
+                                return html
+
+                        Gui.register_content_provider((PlotlyFigure, dict), get_plotly_content)
```

Comment:
that won't work register_content_provider doesn't take a tuple as parameter

Reply:
sorry for making such mistakes. Shall I register the content provider for each type separately?

Reply:
I'm afraid it is more complicated than that
some Map chart don"t use figure at all and are completely built on the frontend
there are not using the content provider mechanism.

Reply:
don't apologize for making mistakes, that's the best way to learn :-)
Thanks for your help

Reply:
thank you for your kind words. what shall I do ahead

Reply:
I propose that you understand how the chart component handle plotly.js charts on the frontend and figure a way to provide the right token from the backend to the frontend

Reply:
okay sir i am on it.